### PR TITLE
Remove homemade optional - use std::optional

### PR DESCRIPTION
**Pull request by joakim-hove**
EclipseGrid used to use a home made class resembling `std::optional<T>`- this PR replaces with `std::optional<>`

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: CMakeLists_files.cmake [hunk 1]
```diff
@@ -376,7 +376,6 @@ if(ENABLE_ECL_INPUT)
     tests/parser/TuningTests.cpp
     tests/parser/UDQTests.cpp
     tests/parser/UnitTests.cpp
-    tests/parser/ValueTests.cpp
```
Comment (atgeirr) [inline CMakeLists_files.cmake:379]: This file should also be deleted.

**Review by atgeirr — APPROVED** [review]
Looks good, I guess the name feature was not essential.

Should be merged when the forgotten file is removed, and green.

File: CMakeLists_files.cmake (hunk 1 above)
Comment (joakim-hove) [reply CMakeLists_files.cmake:379]: Done

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**